api/src/components/AddProduct.tsx

```diff
@@ -0,0 +1,98 @@
+
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import {useEffect, useState} from 'react';
+import axios from 'axios';
+
+export default function FormDialog() {
+  const [open, setOpen] = useState(false);
+  const [title,setTitle] = useState('');
+  const [description, setDescription] = useState('');
+
+const handleDescriptionChange = (event) => {
+  setDescription(event.target.value)
+}
+console.log(description)
+
+  const handleTitleChange = (event) => {
+    setTitle(event.target.value)
+  }
+  console.log(title);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    useEffect(() => {
+     axios.post('https://dummyjson.com/products', {
+      title : title,
+      description : description 
+     }).then((res) => {
+      console.log(res.data); setOpen(false)
+    }).catch((error) => console.error(' Error adding product ' , error))
+    })
+
+  };
+  return (
+    <React.Fragment>
+      <Button variant="outlined" onClick={handleClickOpen} >
+        Create a product
+      </Button>
+      <Dialog
+        open={open}
+        onClose={() => setOpen(false)}
+        PaperProps={{
+          component: 'form',
+          onSubmit: (event: React.FormEvent<HTMLFormElement>) => {
+            event.preventDefault();
+            const formData = new FormData(event.currentTarget);
+            const formJson = Object.fromEntries((formData as any).entries());
+            const email = formJson.email;
```

Comment:
How are you expecting this to work if you are still having the boilerplate from MUI with email?
You should have a handleSubmit function where you handle the operations related to data submissions, this is a logical order of executing things.
In handleSubmit you make a try catch with a axios post where you send to the serve the data that you want to create.